utils/video_demuxer.h

```diff
@@ -327,20 +318,18 @@ class VideoDemuxer {
                 } while (seek_done != 0);
 
                 seek_ctx.out_frame_pts_ = pkt_data.pts;
-                seek_ctx.out_frame_dts_ = pkt_data.dts;
-                seek_ctx.requested_frame_dts_ = timestamp;
                 seek_ctx.out_frame_duration_ = pkt_data.duration = pkt_duration_;
+                seek_ctx.requested_frame_pts_ = (int64_t) (timestamp * default_time_scale_ * time_base_);
             };
 
             // Seek for closest key frame in the past;
             auto seek_for_prev_key_frame = [&](PacketData& pkt_data, VideoSeekContext& seek_ctx) {
                 seek_frame(seek_ctx, AVSEEK_FLAG_BACKWARD);
-                Demux(pp_video, video_size, &pkt_data.pts, &pkt_data.dts);
+                Demux(pp_video, video_size, &pkt_data.pts);
                 seek_ctx.num_frames_decoded_ = static_cast<uint64_t>(pkt_data.pts / 1000 * frame_rate_);
                 seek_ctx.out_frame_pts_ = pkt_data.pts;
-                seek_ctx.out_frame_dts_ = pkt_data.dts;
-                seek_ctx.requested_frame_dts_ = timestamp;
                 seek_ctx.out_frame_duration_ = pkt_data.duration = pkt_duration_;
+                seek_ctx.requested_frame_pts_ = (int64_t) (timestamp * default_time_scale_ * time_base_);
```

Comment:
is requested_frame_pts used?

Reply:
It is being used [here ](https://github.com/fiona-gladwin/rocAL/blob/fbdf4bd352443815a56059df10d59eda795730c1/rocAL/source/decoders/video/rocdec_video_decoder.cpp#L142)in rocAL to discard the other frames until the PTS value returned by decoder is equal to the requested frame PTS